portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,33 +46,46 @@ private String convertToJsonUsingGson(List<String> messages) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get comments in datastore, by most recent order at the top.
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // Iterate over all entities, get comment text.
-    List<String> comments = new ArrayList<>();
+    // Extract limit on number of comments from query string.
+    Integer limit = Integer.parseInt(request.getParameter("limit"));
```

Comment:
you should be able to convert the Integer to an int automatically. As I wrote in the comment below, it's usually better to use the primitive type than the wrapper object when possible.

So you could write "int limit = Integer.ParseInt(...)"